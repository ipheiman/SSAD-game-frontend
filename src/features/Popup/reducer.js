const initialState = {
    //fetch list of question coords from backend
    qlist:  [
            [1, 7],
            [14, 3],
            ],
  };
  
  const popupReducer = (state = initialState, action) => {
    switch (action.type) {
      case "ATTEMPT_Q":
        return {
          //get obstacle coord, remove from qList in store
        };
      default:
        return state;
    }
  };
  
  export default popupReducer;
  