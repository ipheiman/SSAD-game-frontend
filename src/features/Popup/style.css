.popup {
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: auto;
  background-color: rgba(0, 0, 0, 0.5);
}

/*#region Battle-Backgrounds*/
.battle-pikachu {
  position: relative;
  top: 2%;
  width: 600px;
  height: 600px;
  margin: auto;
  border-radius: 20px;
  background-image: url("/battle/battle-pikachu.png");
  background-position: center bottom;
  background-repeat: no-repeat;
  background-color: #464646;
}
.battle-snorlax {
  position: relative;
  top: 2%;
  width: 600px;
  height: 600px;
  margin: auto;
  border-radius: 20px;
  background-image: url("/battle/alive-snorlax.png");
  background-position: center bottom;
  background-repeat: no-repeat;
  background-color: #464646;
}
.battle-bulbasuar {
  position: relative;
  top: 2%;
  width: 600px;
  height: 600px;
  margin: auto;
  border-radius: 20px;
  background-image: url("/battle/battle-bulbasaur.png");
  background-position: center bottom;
  background-repeat: no-repeat;
  background-color: #464646;
}
.battle-charizard {
  position: relative;
  top: 2%;
  width: 600px;
  height: 600px;
  margin: auto;
  border-radius: 20px;
  background-image: url("/battle/battle-charizard.png");
  background-position: center bottom;
  background-repeat: no-repeat;
  background-color: #464646;
}
.battle-clefairy {
  position: relative;
  top: 2%;
  width: 600px;
  height: 600px;
  margin: auto;
  border-radius: 20px;
  background-image: url("/battle/battle-clefairy.png");
  background-position: center bottom;
  background-repeat: no-repeat;
  background-color: #464646;
}
.battle-dratini {
  position: relative;
  top: 2%;
  width: 600px;
  height: 600px;
  margin: auto;
  border-radius: 20px;
  background-image: url("/battle/battle-dratini.png");
  background-position: center bottom;
  background-repeat: no-repeat;
  background-color: #464646;
}
.battle-marill {
  position: relative;
  top: 2%;
  width: 600px;
  height: 600px;
  margin: auto;
  border-radius: 20px;
  background-image: url("/battle/battle-marill.png");
  background-position: center bottom;
  background-repeat: no-repeat;
  background-color: #464646;
}
.battle-seel {
  position: relative;
  top: 2%;
  width: 600px;
  height: 600px;
  margin: auto;
  border-radius: 20px;
  background-image: url("/battle/battle-seel.png");
  background-position: center bottom;
  background-repeat: no-repeat;
  background-color: #464646;
}
.battle-spearow {
  position: relative;
  top: 2%;
  width: 600px;
  height: 600px;
  margin: auto;
  border-radius: 20px;
  background-image: url("/battle/battle-spearow.png");
  background-position: center bottom;
  background-repeat: no-repeat;
  background-color: #464646;
}
/* #endregion */

.popup-header{
  position: relative;
  padding-top: 15px;
  text-align: center;
  color: #d8b9c3;
}
.popup-btn-run{
  position: absolute;
  text-align: center;
  padding: 2px;
  width: 80px;
  height: 30px;
  bottom: 20px;
  right: 80px;
  background-color: beige;
  color: black;
}
