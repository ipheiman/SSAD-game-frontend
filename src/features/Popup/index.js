import React from "react";
import "./style.css";

class Popup extends React.Component {
  render() {
    return (
      <div data-testid="POPUP" className="popup">
        <div data-testid="POPUP_INNER" className="popup_inner">
          <h1 data-testid="TEXT">{this.props.text}</h1>
          <button
            data-testid="BUTTON"
            onClick={this.props.closePopup}
            label="Close me"
          >
            close me
          </button>
        </div>
      </div>
    );
  }
}

export default Popup;
