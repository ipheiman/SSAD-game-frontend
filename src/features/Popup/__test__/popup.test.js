import React, { Component } from "react";
import ReactDOM from "react-dom";
import Popup from "../../Popup";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Popup></Popup>, div);
});

it("should be enabled", () => {
  const { getByTestId } = render(<Popup />);
  expect(getByTestId("BUTTON")).not.toHaveAttribute("disabled");
});

it("should take a snapshot", () => {
  const { asFragment } = render(<Popup />);

  expect(asFragment(<Popup />)).toMatchSnapshot();
});
