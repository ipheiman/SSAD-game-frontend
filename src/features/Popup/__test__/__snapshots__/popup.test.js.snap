// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="popup"
    data-testid="POPUP"
  >
    <div
      class="popup_inner"
      data-testid="POPUP_INNER"
    >
      <h1
        data-testid="TEXT"
      />
      <button
        data-testid="BUTTON"
        label="Close me"
      >
        close me
      </button>
    </div>
  </div>
</DocumentFragment>
`;
